### PR TITLE
[contracts] Add contract for batch validating orders

### PR DESCRIPTION
**Pull request by abandeali1**
## Checklist:

*   [ ] Prefix PR title with `[WIP]` if necessary.
*   [ ] Prefix PR title with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [ ] Add tests to cover changes as needed.
*   [ ] Update documentation as needed.
*   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+
+contract OrderValidator {
+
+    bytes4 constant internal ERC20_DATA_ID = bytes4(keccak256("ERC20Token(address)"));
+    bytes4 constant internal ERC721_DATA_ID = bytes4(keccak256("ERC721Token(address,uint256)"));
+
+    using LibBytes for bytes;
+
+    struct TraderInfo {
```
Comment (BMillman19) [inline packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol]: can we get balance and allowance of ZRX for each of these as well?

Comment (abandeali1) [reply packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol]: Whoops, forgot about that.

File: packages/migrations/src/2.0.0-beta-testnet/migration.ts [hunk 2]
```diff
@@ -75,6 +76,16 @@ export const runV2TestnetMigrationsAsync = async (
     );
     artifactsWriter.saveArtifact(assetProxyOwner);
 
+    // Deploy OrderValidator
```
Comment (BMillman19) [inline packages/migrations/src/2.0.0-beta-testnet/migration.ts]: moving this to the end of the migration script fixes the tests

File: packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        ordersInfo = EXCHANGE.getOrdersInfo(orders);
+        tradersInfo = getTradersInfo(orders, takerAddresses);
+        return (ordersInfo, tradersInfo);
+    }
+
+    /// @dev Fetches balance and allowances for maker and taker of order.
+    /// @param order The order structure.
+    /// @param takerAddress Address that will be filling the order.
+    /// @return Balances and allowances of maker and taker of order.
+    function getTraderInfo(LibOrder.Order memory order, address takerAddress)
```
Comment (dekz) [inline packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol]: Might be some unexpected behaviour if order has specified a taker address which doesn't match the provided address.

Comment (abandeali1) [reply packages/contracts/src/2.0.0/extensions/OrderValidator/OrderValidator.sol]: We could special case that here, but it's probably easy enough to do off-chain.

**Pull request merged**